examples/bigcommerce-catalyst/components/product-carousel/product-carousel.makeswift.ts

```diff
@@ -0,0 +1,46 @@
+import dynamic from 'next/dynamic';
+
+import { Combobox, TextInput } from '@makeswift/runtime/controls';
+import { forwardNextDynamicRef } from '@makeswift/runtime/next';
+
+import { runtime } from '~/lib/makeswift/runtime';
+
+type Category = {
+  entityId: number;
+  name: string;
+  path: string;
+}
+
+export const props = {
+  title: TextInput({ label: 'Title', defaultValue: 'Products', selectAll: true }),
+  products: Combobox({
+    label: "Category",
+    async getOptions(query) {
+      const { categories }: { categories: Category[] } = await fetch('/api/categories').then((res) => res.json())
+
+      const mappings = await Promise.all(categories
+        .map(async (category) => {
+          const { products } = await fetch(`/api/quick-search?categoryId=${category.entityId}&imageHeight=500&imageWidth=500&limit=10`).then((res) => res.json())
```

Comment:
BC has some limited GraphQL endpoints so this was the monstrosity I had to come up with unfortunately.